### PR TITLE
roachtest: upgrade to/from 23.2 in mixed-version tests

### LINKED ISSUE #115425: roachtest: prepare for 23.2 mixed-version tests
We're not quite ready to make 23.2 master's predecessor yet (even though it should be). Notably, the following need to be addressed:

* There's a bug computing the cockroach URL for beta releases in `mixedversion` tests. See current failure in #115393.
* We need to mint 23.2.
* We need 23.2 fixtures.

Jira issue: CRDB-34027

### PR DESCRIPTION
**Pull request by renatolabs**
**release: support pre-releases in `PredecessorHistory`**
There was a bug computing the list of active releases within a series:
the previous logic would not take pre-releases into account, and would
just iterate over the patch releases instead. As a result, if the
latest release in a series is a pre-release, the code would
incorrectly conclude that the only existing release for that series
was the `.0` release.

In this commit, we handle the case when the latest release is a
pre-release and use that instead.

**roachtest: remove old fixture files**
Fixtures older than `v21.2` were removed. There are no tests currently
using them.

**roachtest: add v23.2 fixture files**
Generated with 23.2.0-rc.2.

**testutils: update predecessor release information**
Now that we have fixtures for 23.2, we are able to update the
predecessors file and run mixed-version tests that step through 23.2
while upgrading to master (24.1).

We also skip logictests that fail in this commit, now that the
cockroach-go config will update from 23.2 to master. These tests will
be handled separately by the owning teams.

Fixes: #115425.

Release note: None

### CODE REVIEW

**Comment by renatolabs** [comment]
@cockroachdb/test-eng this is now ready for review pls 🙏

File: pkg/testutils/release/releases_test.go [hunk 1]
```diff
@@ -61,8 +68,8 @@ func TestLatestAndRandomPredecessor(t *testing.T) {
 	}{
 		{
 			name:        "non-existent release",
-			v:           "v23.2.3",
-			expectedErr: `no release information for "v23.2.3" ("23.2" series)`,
+			v:           "v50.8.3",
```
Comment (srosenberg) [inline pkg/testutils/release/releases_test.go:71]: Nice, we'll need to update this in 2050!

**Review by srosenberg — APPROVED** [review]
Thanks for adding the new fixtures and cleaning up the old ones!

File: pkg/testutils/release/releases_test.go (hunk 1 above)
Comment (renatolabs) [reply pkg/testutils/release/releases_test.go:71]: Hehe, not really, this test is reading from `testReleaseData`, not from the real YAML file. I added a 23.2 entry in the test data to test the changes to support pre-releases, that's why I needed a new non-existent version in this test.

**Comment by renatolabs** [comment]
TFTR!

bors r=srosenberg

**Pull request merged**